packages/utils/src/commonwell/migrate-patients.ts

```diff
@@ -0,0 +1,149 @@
+ * 3. Run the script with `ts-node src/commonwell/migrate-patients.ts`
+ */
+
+dayjs.extend(duration);
+
+const patientIds: string[] = [];
+const rerunPdOnNewDemographics = false;
+const cxId = getEnvVarOrFail("CX_ID");
+const apiUrl = getEnvVarOrFail("API_URL");
+
+const cwOid = getEnvVarOrFail("CW_OID");
+const cwCert = getEnvVarOrFail("CW_CERT");
+const cwKey = getEnvVarOrFail("CW_KEY");
+const apiMode = APIMode.integration;
+
+const PATIENT_CHUNK_SIZE = 2;
+
+const oldCwOid = "";
+const cwOrgName = "";
+
+const confirmationTime = dayjs.duration(10, "seconds");
+const delayTime = dayjs.duration(10, "seconds");
+
+type PdResponse = {
+  data: {
+    requestId: string;
+  };
+};
+
+function buildCWPatientId(orgOID: string, patientId: string): string {
+  return `${patientId}%5E%5E%5Eurn%3aoid%3a${orgOID}`;
+}
+
+async function main() {
+  try {
+    await displayInitialWarningAndConfirmation(patientIds.length);
+
+    const patientChunks = chunk(patientIds, PATIENT_CHUNK_SIZE);
+    const commonWell = new CommonWell(cwCert, cwKey, cwOrgName, oldCwOid, apiMode);
+
+    let successCount = 0;
+    let failureCount = 0;
+    const failures: Record<string, string> = {};
+
+    for (const [i, patients] of patientChunks.entries()) {
+      console.log(`Chunk ${i + 1} of ${patientChunks.length}`);
+      console.log(`# of patients ${patients.length}`);
+
+      for (const patientId of patients) {
+        const log = out(`Patient migration: cxId - ${cxId}, patientId - ${patientId}`).log;
+        try {
+          log("Resetting external data...");
+          const resetUrl = `${apiUrl}/internal/patient/${patientId}/external-data`;
+          await axios.put(resetUrl, null, {
+            params: {
+              cxId,
+              source: MedicalDataSource.COMMONWELL,
+            },
+          });
+
+          log("Running patient discovery...");
+          const pdUrl = `${apiUrl}/internal/commonwell/patient-discovery/${patientId}`;
+          const pdParams = new URLSearchParams({
+            cxId,
+            rerunPdOnNewDemographics: rerunPdOnNewDemographics.toString(),
+          });
+          const resp = (await axios.post(`${pdUrl}?${pdParams}`)) as PdResponse;
+          log(`Request ID - ${JSON.stringify(resp.data.requestId)}`);
+
+          // 3. Delete patient in CommonWell
+          log("Deleting patient in CommonWell...");
+          const queryMeta = organizationQueryMeta(cwOrgName, {
+            npi: addOidPrefix(oldCwOid),
+          });
+
+          const cwPatientId = buildCWPatientId(oldCwOid, patientId);
+
+          await commonWell.deletePatient(queryMeta, cwPatientId);
+
+          log("Patient migration completed successfully");
+          successCount++;
+        } catch (err) {
+          log(`Failed to migrate patient: ${err}`);
+          failures[patientId] = String(err);
+          failureCount++;
+        }
+      }
+
+      if (i < patientChunks.length - 1) {
+        const sleepTime = delayTime.asMilliseconds();
```

Comment:
Let's use `getDelayTime()` so we can change it dynamically while we're executing this.

Reply:
Interesting util, makes sense. If you could add a note to the tsdoc at the top about needing to have an appropriately named file in the directory you're invoking the script from that would be plus.

Reply:
If you could add a note to the tsdoc at the top about needing to have an appropriately named file in the directory you're invoking the script from that would be plus.

@lucasdellabella i dont follow?

Reply:
It will be confusing if the script fails because a random file doesn't exist. imo for people new to oncall, its nice to put something in the tsdoc at the top to explain what setup they need to do to have the script run successfully.

Reply:
There is a tsdoc at the top stating how to set it up and also what file wouldnt exist?